apps/nextjs/src/app/additional-material-spike/AdditionalMaterials.tsx

```diff
@@ -0,0 +1,313 @@
+    "subject",
+    "topic",
+    "learningOutcome",
+    "learningCycles",
+    "priorKnowledge",
+    "keyLearningPoints",
+    "misconceptions",
+    "keywords",
+    "starterQuiz",
+    "cycle1",
+    "cycle2",
+    "cycle3",
+    "exitQuiz",
+    "additionalMaterials",
+  ];
+
+  return sectionKeys.map((key) => ({
+    key,
+    data: lessonPlan[key as keyof typeof lessonPlan] ?? null,
+  }));
+}
+
+export type Cycle = {
+  title: string;
+  durationInMinutes: number;
+  explanation: {
+    imagePrompt: string;
+    spokenExplanation: string[];
+    accompanyingSlideDetails: string;
+    slideText: string;
+  };
+  practice: string;
+  feedback: string;
+} | null;
+
+interface AdditionalMaterialsProps {
+  pageData: {
+    lessonPlan: AilaPersistedChat["lessonPlan"];
+    transcript?: string | null;
+  };
+}
+
+const AdditionalMaterials: FC<AdditionalMaterialsProps> = ({ pageData }) => {
+  const [generation, setGeneration] = useState<AdditionalMaterialType | null>(
+    null,
+  );
+  const [moderation, setModeration] = useState<string | null>(null);
+  const [action, setAction] = useState<string | null>(null);
+  const fetchMaterialModeration =
+    trpc.additionalMaterials.getAdditionalMaterialModeration.useMutation();
+  const fetchMaterial =
+    trpc.additionalMaterials.getAdditionalMaterial.useMutation();
+  const [prompt, setPrompt] = useState<{
+    prompt: string;
+    systemMessage: string;
+  } | null>(null);
+
+  const canSeeSpike = useClientSideFeatureFlag("additional-materials");
+
+  if (!canSeeSpike) {
+    redirect("/");
+  }
+
+  useEffect(() => {
+    if (action) {
+      const prompt = additionalMaterialsConfig[action]?.getPrompt({
+        lessonPlan: pageData.lessonPlan,
+      });
+      setPrompt(
+        prompt as {
+          prompt: string;
+          systemMessage: string;
+        },
+      );
+    }
+  }, [action, pageData.lessonPlan]);
+
+  useEffect(() => {
+    if (
+      generation !== null &&
+      !fetchMaterialModeration.isLoading &&
+      !moderation
+    ) {
+      const fetchData = async () => {
+        try {
+          const data = await fetchMaterialModeration.mutateAsync({
+            generation: JSON.stringify(generation),
+          });
+
+          setModeration(data);
+        } catch (error) {
+          log.error("error", error);
+        }
+      };
+      void fetchData();
+    }
+  }, [fetchMaterialModeration, generation, moderation]);
+
+  const handleSubmit = async (message?: string) => {
+    setGeneration(null);
+    if (!action) {
+      alert("No action selected");
+      return;
+    }
+
+    try {
+      const res = await fetchMaterial.mutateAsync({
+        lessonPlan: pageData.lessonPlan,
+        action: action,
+        message: message ?? undefined,
+        previousOutput: generation,
+      });
+
+      setGeneration(res);
+    } catch (error) {
+      log.error("error", error);
+    }
+  };
+  const renderGeneratedMaterial = () => {
+    if (!generation) return null;
```

Comment:
My usual question about early returns: Should this be an invariant? Do you expect it to happen in typical use?